zlib-rs/src/deflate.rs

```diff
@@ -1281,6 +1283,11 @@ pub(crate) struct State<'a> {
     /// than this value. This mechanism is used only for compression levels >= 4.
     pub(crate) max_lazy_match: u16,
 
+    /// number of string matches in current block
+    /// Note: this counter is just 8 bits to help keep the struct compact. Code that
+    /// increments it must be careful to avoid overflow.
+    pub(crate) matches: u8,
```

Comment:
Is there reasoning behind why 8 bits is sufficient? Either that's enough and you don't need to be careful, or it's not and then we should make this a `u16`?

Reply:
I revised the comment to explain why an 8 bit saturating counter is sufficient. Basically, the code only uses it to differentiate the `< 2` and `> 2` cases, so a 2 bit saturating counter would work. This leaves 8 bits of unused space in the structure after the `matches` field; I may be able to move something else there in a future PR.